### ISSUE TITLE
I get some errors (i probably did something wrong)

### ISSUE THREAD

**Issue by S3RL4** [SociallyIneptWeeb/LanguageLeapAI#3]
How do i fix these errors? Do i have some incompatible version of python or something?

![image](https://user-images.githubusercontent.com/92375490/224327711-518b1f8e-770f-4b97-9a59-f2bade0959f8.png)

**Comment by SociallyIneptWeeb** [comment]
You might have to ensure that Docker Desktop is running

**Comment by S3RL4** [comment]
> You might have to ensure that Docker Desktop is running, see this [link](https://stackoverflow.com/questions/57108228/how-to-check-if-docker-is-running-on-windows)

the download link for docker desktop did not work for me, so i installed it via pip.
Also i only have docker-compose installed

**Comment by SociallyIneptWeeb** [comment]
Are you on Windows/Mac/Linux?

**Comment by S3RL4** [comment]
> 
Windows 11 dev channel build

**Comment by SociallyIneptWeeb** [comment]
I think you can try uninstalling your current docker compose and follow the instructions [here](https://docs.docker.com.zh.xy2401.com/v17.12/docker-for-windows/install/).

**Comment by S3RL4** [comment]
> I think you can try uninstalling your current docker compose and follow the instructions [here](https://docs.docker.com.zh.xy2401.com/v17.12/docker-for-windows/install/).

yeah, but the download link doesnt work for windows stable nor the edge version.

**Comment by SociallyIneptWeeb** [comment]
Oooo I might have put the wrong link then, use [this one](https://docs.docker.com/desktop/install/windows-install/) instead

**Comment by S3RL4** [comment]
> Oooo I might have put the wrong link then, use [this one](https://docs.docker.com/desktop/install/windows-install/) instead

well now we are talking xd
lemme try running it with that one, and i'll get back to you with some updates on the situation

**Comment by SociallyIneptWeeb** [comment]
That was my bad for putting the wrong link, thanks for creating an issue!

**Comment by S3RL4** [comment]
No problem. You replying to issues here so quickly alone makes up for it xd

**Comment by S3RL4** [comment]
Np again xd. Could you hop in a dc call to explain some stuff for a few mins? (might be too much to ask for but you can never know)

**Comment by SociallyIneptWeeb** [comment]
I will probably make a video for the setup of this, I'm not too comfortable with using discord as of right now, so if you have any further questions you can ask it here or in a new issue :)

**Comment by S3RL4** [comment]
oh ok. it's just that this issue thread might get pretty long and some things might be hard to explain here.

how do i know if it's running in some way? cuz nothing seems to happen after i did that.
![image](https://user-images.githubusercontent.com/92375490/224338098-e9d4b91c-ef88-4591-9848-429477dcb82e.png)

**Comment by SociallyIneptWeeb** [comment]
You can try holding down the f key, cos right now the default push to talk key is the f key. Only after holding down that key will the program record your mic if you have set up the audio device IDs in your .env file correctly.

**Comment by S3RL4** [comment]
nothing seems to happen, but i did have a suspicion on my device IDs from the start.

![image](https://user-images.githubusercontent.com/92375490/224338901-7cc6a71d-0ed5-4fb9-bf86-65b78711162a.png)
can you try to explain in more detail how those are supposed to be put there?

**Comment by SociallyIneptWeeb** [comment]
You will have to run src/modules/get_audio_devices_id.py. 
 Using the output, you can assign the correct IDs to the env variables.

**Comment by S3RL4** [comment]
> You will have to run src/modules/get_audio_devices_id.py. Using the output, you can assign the correct IDs to the env variables.

yeah i figured that but do i have to put the names with the number IDs? or just the stuff after the number?
![image](https://user-images.githubusercontent.com/92375490/224339741-cfa61d9b-5acb-4c40-9d7f-bf29b9a706a8.png)

**Comment by S3RL4** [comment]
> So if the mic u r using is Scarlett solo usb, you will edit the .env file to MICROPHONE_ID=1. Only the id is required. The names are there so you know which id refers to which audio device.

that's exactly how my env thing was done when i ran the python program tho.

**Comment by SociallyIneptWeeb** [comment]
But when you hold down the push to talk key nothing happens? No output at all? There's worrying, maybe trying adding some print statements to the on_press_key and on_release_key functions in voice_translator.py and see if there is any output at all when you hold down the push to talk key.

**Comment by S3RL4** [comment]
i feel like it might have something to do with my voicemeeter or maybe i still have the wrong id selected since the device id list has 71 devices and there seems to be duplicate ones
![image](https://user-images.githubusercontent.com/92375490/224342375-d089bf6d-af54-4504-bb94-1127bcdea068.png)

![image](https://user-images.githubusercontent.com/92375490/224342413-df8d26d1-6ea3-41ff-acf8-d0b2bd7a279f.png)

![image](https://user-images.githubusercontent.com/92375490/224342472-d6672dee-3caa-4c0d-aba9-f46327f0e0ae.png)


that's how my dc test setup with it
![image](https://user-images.githubusercontent.com/92375490/224342502-174b8840-a71b-4cff-8729-eb82537b8888.png)

**Comment by SociallyIneptWeeb** [comment]
The duplicate audio devices are ok, it will just start repeating itself for some weird reason but otherwise the ids should be correct.

**Comment by S3RL4** [comment]
> The duplicate audio devices are ok, it will just start repeating itself for some weird reason but otherwise the ids should be correct.

do you see any problems that stick out to you from those screenshots though?

**Comment by SociallyIneptWeeb** [comment]
Not exactly, I have a feeling it might be the python program not capturing your keyboards input. Even if it doesnt receive any audio, it should still print (no speech detected) if the on_key_press and on_key_release functions are being called correctly.

**Comment by S3RL4** [comment]
> 

i ran the script via git bash. could it be cuz of that?

**Comment by SociallyIneptWeeb** [comment]
ooo maybe, try running in your own cmd

**Comment by S3RL4** [comment]
> ooo maybe, try running in your own cmd

it does something in cmd but all i see is errors xd
![image](https://user-images.githubusercontent.com/92375490/224345746-9e34e1bf-0934-408e-acc1-6c4e28223241.png)

**Comment by SociallyIneptWeeb** [comment]
Nice this is good, now you just have to ensure that your voicevox and whisper docker containers are running

**Comment by S3RL4** [comment]
> 

everything there was running fine till i ran the code and now whisperai is in exited state
![image](https://user-images.githubusercontent.com/92375490/224346717-dd220728-d334-4b53-a6db-ecc45ac8aec0.png)

**Comment by SociallyIneptWeeb** [comment]
Ooo yikes, maybe try removing the containers, deleting the images and running docker compose up again

**Comment by S3RL4** [comment]
> Ooo yikes, maybe try removing the containers, deleting the images and running docker compose up again

yeah i restarted and now it seems to be working
![image](https://user-images.githubusercontent.com/92375490/224347942-357b4a7e-e605-41e2-b4c0-427202db3d7f.png)
i can hear the voice too and just got ear raped by that thing xddd
not sure if it works in dc or other programs yet tho

**Comment by S3RL4** [comment]
now it works nvm. it seems that sometimes the program can freak out and not work properly but restarting it / or everything if necessary solves the problems.

**Comment by SociallyIneptWeeb** [comment]
Alright good to know 👍

**Comment by S3RL4** [comment]
i think i'll mark this as closed cuz as far as i know everything works.
Thanks for the help xd
